addons/sale_gift_card/models/sale_order.py

```diff
@@ -50,6 +50,15 @@ def _pay_with_gift_card(self, gift_card):
             })
         return error
 
+
+    def _get_amount_paid_with_gift_card(self):
+        gift_card_lines = self.env['sale.order.line'].search([
+            ('order_id', '=', self.id),
+            ('gift_card_id', '!=', None),
+        ])
+        return super()._get_amount_paid_with_gift_card() + sum(gift_card_lines.mapped('price_unit'))
```

Comment:
```suggestion
    def _compute_amount_total_without_delivery(self):
        self.ensure_one()
        # Add back 'payment' rewards from the total without delivery, they should count towards the delivery price goal.
        lines = self.order_line.filtered(lambda l: l.reward_id.program_id.is_payment_program)
        return super()._get_amount_paid_with_gift_card() + sum(lines.mapped('price_unit'))
```

Reply:
It's a bit tricky how the override is used there but I think I understand